packages/boot/test/upgrading/upgrade-vats.test.js

```diff
@@ -562,22 +557,11 @@ test('upgrade vat-vow', async t => {
     vowFulfilled: { status: 'fulfilled', value: 'hello' },
     vowRejected: { status: 'rejected', reason: 'goodbye' },
     vowPostUpgrade: { status: 'fulfilled', value: 'bonjour' },
-    vowExternalPromise: { status: 'fulfilled', value: 'ciao' },
-    vowExternalVow: {
-      status: 'rejected',
-      reason: {
-        name: 'vatUpgraded',
-        upgradeMessage: 'vat upgraded',
-        incarnationNumber: 0,
-      },
-    },
-    vowPromiseForever: {
-      status: 'rejected',
-      reason: {
-        name: 'vatUpgraded',
-        upgradeMessage: 'vat upgraded',
-        incarnationNumber: 0,
-      },
-    },
+    // The 'fulfilled' result below is wishful thinking.  Long-lived
+    // promises are not supported by `watch` at this time.
+    // vowExternalPromise: { status: 'fulfilled', value: 'ciao' },
```

Comment:
Let's discuss this more. I think we can handle this case. Unlike what the comment states, `watch` can already handle externally decided promises (I can add a test for that if you want if there isn't one). The problem is that the internals of vow do not directly watch the resolution value if it's a promise, and instead adopt it into a newly created promise kit.

Reply:
Fixed!

Reply:
I have to admit, I don't fully grasp the different states of promise vow and the resolver kit logic, but I trust it sufficiently works for not given the tests.